stats/opentelemetry/e2e_test.go

```diff
@@ -276,27 +271,35 @@ func validateTraces(t *testing.T, spans tracetest.SpanStubs, wantSpanInfos []tra
 		}
 	}
 
-	// Constructs a map from a slice of traceSpanInfo to retrieve the
-	// corresponding expected span info based on span name and span kind
-	// for comparison.
-	wantSpanInfosMap := make(map[traceSpanInfoMapKey]traceSpanInfo)
-	for _, info := range wantSpanInfos {
-		key := traceSpanInfoMapKey{spanName: info.name, spanKind: info.spanKind}
-		wantSpanInfosMap[key] = info
-	}
+	// Sort wantSpanInfos by name and kind for deterministic ordering.
+	sort.Slice(wantSpanInfos, func(i, j int) bool {
+		if wantSpanInfos[i].name == wantSpanInfos[j].name {
+			return wantSpanInfos[i].spanKind < wantSpanInfos[j].spanKind
+		}
+		return wantSpanInfos[i].name < wantSpanInfos[j].name
+	})
+
+	// Copy spans and sort by name and kind.
+	sortedSpans := make([]tracetest.SpanStub, len(spans))
+	copy(sortedSpans, spans)
+	sort.Slice(sortedSpans, func(i, j int) bool {
+		if sortedSpans[i].Name == sortedSpans[j].Name {
+			return sortedSpans[i].SpanKind.String() < sortedSpans[j].SpanKind.String()
+		}
+		return sortedSpans[i].Name < sortedSpans[j].Name
+	})
 
 	// Compare retrieved spans with expected spans.
-	for _, span := range spans {
+	for i := range sortedSpans {
```

Comment:
Can we run `cmp.Diff` on `wantSpanInfosMap` and `sortedSpans` directly instead?

To do this you would need to convert `sortedSpans` to `SpanStub` once. Then you can replace the calls to `sort.Slice` with `cmpopts.SortSlices` options passed to `cmp.Diff`.